perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -370,6 +370,27 @@ sub visible_states_remove {
     }
 }
 
+sub public_asset_id {
+    my $self = shift;
+
+    my $current_cobrand = $self->result_source->schema->cobrand;
+    my $cobrand_for_problem
+        = $current_cobrand->get_body_handler_for_problem($self);
```

Comment:
It's defined in FixMyStreet::Cobrand::Base; it calls the method in the cobrand if it exists, and returns undef if it does not. So it means you don't have to define defaults for everything to prevent errors, they'll only be called if they do exist.